### PR TITLE
storage: add gzip option

### LINKED ISSUE #751: Is gzip uploading possible through the API
Hi guys

I have been reading the documentation and from what I understand there is functionality to gzip uploaded files to storage - or at least the functionality exists through the gsutil.

"Note also that gsutil provides an easy way to cause content to be compressed and stored with Content-Encoding: gzip: see the -z option"

Is this currently possible through the node API? I have not seen it mentioned in the docs or code comments. If not, does this mean we have to gzip the files ourselves and then upload it with gzip meta data?

### LINKED ISSUE #752: Show gzip example within docs
Related https://github.com/GoogleCloudPlatform/gcloud-node/issues/751

### PR DESCRIPTION
**Pull request by stephenplusplus**
Fixes #751 
Fixes #752 

This adds automatic gzipping to file uploads:

``` js
// With upload:
bucket.upload('./input.html', { gzip: true }, function(err, file) {
  // file was gzipped!
  // file.metadata.contentEncoding = 'gzip'
});

// Or with streams:
var file = bucket.file('input.html');
var fs = require('fs');

fs.createReadStream('./input.html')
  .pipe(file.createWriteStream({ gzip: true }))
  .on('complete', function(metadata) {
     // file was gzipped!
     // metadata.contentEncoding = 'gzip'
  });
```

### CODE REVIEW

File: lib/storage/bucket.js [hunk 1]
```diff
@@ -927,6 +929,20 @@ Bucket.prototype.setMetadata = function(metadata, callback) {
  * });
  *
  * //-
+ * // You can also have a file gzip'd on the fly.
+ * //-
+ * var fs = require('fs');
```
Comment (callmehiphop) [inline lib/storage/bucket.js]: These don't look like they're being used within the examples..?

Comment (stephenplusplus) [inline lib/storage/bucket.js]: Woops.

Comment (stephenplusplus) [inline lib/storage/bucket.js]: Fixed.

**Comment by callmehiphop** [comment]
Looks good to me!

**Pull request merged**